### PR TITLE
RUBY-1559 Update URI option spec tests and fix comparisons

### PR DESCRIPTION
**Pull request by saghm**
The URI options have been merged, so this PR includes the changes that have been made since the initial commit for them.

Additionally, it appears that the test runner wasn't properly comparing the expected URI options to their parsed options of different names, so I've fixed that as well.

### CODE REVIEW

File: lib/mongo/uri.rb [hunk 1]
```diff
@@ -455,8 +455,7 @@ def self.uri_option(uri_key, name, extra = {})
     uri_option 'tlsallowinvalidcertificates', :ssl_verify, :type => :ssl_verify
     uri_option 'tlscafile', :ssl_ca_cert
     uri_option 'tlscertificatekeyfile', :ssl_cert
-    uri_option 'tlsclientkeyfile', :ssl_key
-    uri_option 'tlscertificatekeypassword', :ssl_key_pass_phrase
+    uri_option 'tlscertificatekeyfilepassword', :ssl_key_pass_phrase
```
Comment (p-mongo) [inline lib/mongo/uri.rb]: Per https://jira.mongodb.org/browse/RUBY-1676 I'm thinking this change should also have an accompanying documentation change, no?

Comment (saghm) [reply lib/mongo/uri.rb]: Done

File: lib/mongo/uri.rb [hunk 2]
```diff
@@ -647,6 +646,37 @@ def zlib_compression_level(value)
       nil
     end
 
+    # Parses the max pool size.
+    #
+    # @param value [ String ] The max pool size string.
+    #
+    # @return [ Integer | nil ] The min pool size if it is valid, otherwise nil (and a warning will)
```
Comment (p-mongo) [inline lib/mongo/uri.rb]: The closing paren here is a bit too early in the sentence.

Comment (saghm) [reply lib/mongo/uri.rb]: Oops! Good catch

**Pull request merged**